docs/modules/ROOT/pages/how-tos/exoscale/change_storage_node_size.adoc

```diff
@@ -0,0 +1,494 @@
+    ceph osd crush remove "${node}"
+done
+----
+
+. Check that the OSDs are no longer listed in `ceph osd tree`
++
+[source,bash]
+----
+kubectl --as=cluster-admin -n syn-rook-ceph-cluster exec -it deploy/rook-ceph-tools -- \
+  ceph osd tree
+----
+
+. Scale down the Rook-Ceph operator
++
+[source,bash]
+----
+kubectl --as=cluster-admin -n syn-rook-ceph-operator scale --replicas=0 \
+  deploy/rook-ceph-operator
+----
+
+. Make a note of the PVC of the old OSD
++
+NOTE: We also extract the name of the PV here, but we'll only delete the PV after removing the node from the cluster.
++
+[source,bash]
+----
+old_pvc_names=""
+old_pv_names=""
+for node in $(echo -n $NODES_TO_REPLACE); do
+  osd_id=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get deploy \
+    -l failure-domain="${node}" --no-headers \
+    -o custom-columns="NAME:.metadata.name" | cut -d- -f4)
+
+  pvc_name=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get deploy \
+    "rook-ceph-osd-${osd_id}" -ojsonpath='{.metadata.labels.ceph\.rook\.io/pvc}')
+  pv_name=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get pvc \
+    "${pvc_name}" -o jsonpath='{.spec.volumeName}')
+
+  old_pvc_names="$old_pvc_names $pvc_name"
+  old_pv_names="$old_pv_names $pv_name"
+done
+echo $old_pvc_names
+echo $old_pv_names
+----
+
+. Delete old OSD deployments
++
+[source,bash]
+----
+for node in $(echo -n $NODES_TO_REPLACE); do
+  kubectl --as=cluster-admin -n syn-rook-ceph-cluster delete deploy \
+    -l failure-domain="${node}"
+done
+----
+
+. Reset Ceph cluster resource to have original number of OSDs
++
+[source,bash]
+----
+kubectl --as=cluster-admin -n syn-rook-ceph-cluster patch cephcluster cluster --type=json \
+  -p "[{
+    \"op\": \"replace\",
+    \"path\": \"/spec/storage/storageClassDeviceSets/0/count\",
+    \"value\": ${orig_osd_count}
+  }]"
+----
+
+. Clean up PVC and prepare job of the old OSD if necessary
++
+[source,bash]
+----
+for pvc_name in $(echo -n $old_pvc_names); do
+  kubectl --as=cluster-admin -n syn-rook-ceph-cluster delete job \
+    -l ceph.rook.io/pvc="${pvc_name}"
+  kubectl --as=cluster-admin -n syn-rook-ceph-cluster delete pvc "${pvc_name}"
+done
+----
+
+. Clean up PVC encryption secret
++
+[source,bash]
+----
+for pvc_name in $(echo -n $old_pvc_names); do
+  kubectl --as=cluster-admin -n syn-rook-ceph-cluster delete secret -l pvc_name="${pvc_name}"
+done
+----
+
+. Scale up the Rook-Ceph operator
++
+[source,bash]
+----
+kubectl --as=cluster-admin -n syn-rook-ceph-operator scale --replicas=1 \
+  deploy/rook-ceph-operator
+----
+
+=== Remove the old MON
+
+. Find the MON (if any) on the node to be replaced
++
+[source,bash]
+----
+MON_IDS=""
+for node in $(echo -n $NODES_TO_REPLACE); do
+  mon_id=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get pods \
+    -lapp=rook-ceph-mon -o wide \
+    | grep "${node}" | cut -d- -f4)
+  if [[ -n "$mon_id" ]]; then
+    MON_IDS="$MON_IDS $mon_id"
+  fi
```

Comment:
I don't think that is necessary? This list is only read by humans so an extra space shouldn't matter.

Reply:
You're right. I tried automating it first but it's a tad risky. I removed them all at once.